contracts/pools/PoolCollection.sol

```diff
@@ -1619,7 +1627,7 @@ contract PoolCollection is IPoolCollection, Owned, BlockNumber, Utils {
                 revert InsufficientSourceAmount();
```

Comment:
Coincidentally noticed this check, and it feels that either the comment is wrong, or the actual condition is wrong.

Reply:
Hmm, this wasn't a part of the code changes in this PR

Reply:
True, but let's just verify that there's no issue here.

Reply:
It seems correct - 
this is a trade by target amount, and when calculating the estimated source amount needed to make the trade, we check if it's 0 (so source balance or fullTargetAmount is 0 in the calculation) or more than the initially provided `maxSourceAmount` to `tradeByTargetAmount` function (so user hasn't set enough `maxSourceAmount` to get that target amount).